### PR TITLE
use custom connection string

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

use custom connection string - in the connection string we will use the app gateway url
"storage-reform-connection-string" value added to keyvault. 

flux change: https://github.com/hmcts/cnp-flux-config/pull/2078

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/config/StorageConfiguration.java [hunk 1]
```diff
@@ -32,11 +32,10 @@ public LeaseClientProvider getLeaseClientProvider() {
     @Bean
     public BlobServiceClient getStorageClient(
         StorageSharedKeyCredential credentials,
-        @Value("${storage.url}") String storageUrl
+        @Value("${storage.reform.connection-string}") String connectionString
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/config/StorageConfiguration.java]: how will that work for all the other environments?

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/config/StorageConfiguration.java]: for aat and  prod we will use external url which will be bootstrap from vault

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/config/StorageConfiguration.java]: in that case, we need all the usual recipes for value: chart versions, values template for all envs, flux config 😉

File: charts/reform-scan-blob-router/values.preview.template.yaml [hunk 2]
```diff
@@ -41,6 +41,7 @@ java:
     STORAGE_BULKSCAN_URL: https://bulkscanaat.blob.core.windows.net
 
     STORAGE_URL: "https://$(STORAGE_ACCOUNT_NAME).blob.core.windows.net"
+    STORAGE_REFORM_CONNECTION_STRING: "DefaultEndpointsProtocol=https;BlobEndpoint=https://$(STORAGE_ACCOUNT_NAME).blob.core.windows.net;AccountName=$(STORAGE_ACCOUNT_NAME);AccountKey=$(STORAGE_ACCOUNT_KEY);"
```
Comment (nitinprabhu) [inline charts/reform-scan-blob-router/values.preview.template.yaml]: do you intend to use internal or app gw url ?

Comment (mdayican) [reply charts/reform-scan-blob-router/values.preview.template.yaml]: for  pr  it is internal

Comment (doncem) [reply charts/reform-scan-blob-router/values.preview.template.yaml]: aye, `values.yaml` should use app gw whereas preview is fine here

File: src/main/java/uk/gov/hmcts/reform/blobrouter/config/StorageConfiguration.java (hunk 1 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/config/StorageConfiguration.java]: I will add values to vault and revert code changes and release this.

File: charts/reform-scan-blob-router/values.preview.template.yaml (hunk 2 above)
Comment (nitinprabhu) [reply charts/reform-scan-blob-router/values.preview.template.yaml]: `STORAGE_REFORM_CONNECTION_STRING` nothing in values yaml ?

Comment (mdayican) [reply charts/reform-scan-blob-router/values.preview.template.yaml]: - storage-reform-connection-string

File: charts/reform-scan-blob-router/values.yaml [hunk 3]
```diff
@@ -16,6 +16,7 @@ java:
         - crime-storage-connection-string
         - flyway-password
         - blob-router-POSTGRES-PASS
+        - storage-reform-connection-string
```
Comment (doncem) [inline charts/reform-scan-blob-router/values.yaml]: still missing value in prod vault

Comment (mdayican) [reply charts/reform-scan-blob-router/values.yaml]: @nitinprabhu  will add

Comment (mdayican) [reply charts/reform-scan-blob-router/values.yaml]: I added

**Pull request merged**